src/Analyzers/Core/CodeFixes/UseConditionalExpression/ForAssignment/AbstractUseConditionalExpressionForAssignmentCodeFixProvider.cs

```diff
@@ -104,19 +104,25 @@ protected override async Task FixOneAsync(
     private void ConvertOnlyIfToConditionalExpression(
         SyntaxEditor editor,
         IConditionalOperation ifOperation,
-        ISimpleAssignmentOperation assignment,
+        ISimpleAssignmentOperation assignmentOperation,
         TExpressionSyntax conditionalExpression)
     {
         var generator = editor.Generator;
         var ifStatement = (TIfStatementSyntax)ifOperation.Syntax;
-        var expressionStatement = (TStatementSyntax)generator.ExpressionStatement(
-            generator.AssignmentStatement(
-                assignment.Target.Syntax,
-                conditionalExpression)).WithTriviaFrom(ifStatement);
+        var assignment = generator.AssignmentStatement(assignmentOperation.Target.Syntax, conditionalExpression);
+
+        if (assignmentOperation.Parent is IConditionalAccessOperation conditionalAccess)
+        {
+            assignment = generator.ConditionalAccessExpression(
+                conditionalAccess.Operation.Syntax,
+                assignment);
+        }
+
+        var expressionStatement = (TStatementSyntax)generator.ExpressionStatement(assignment);
 
         editor.ReplaceNode(
             ifOperation.Syntax,
-            WrapWithBlockIfAppropriate(ifStatement, expressionStatement));
+            WrapWithBlockIfAppropriate(ifStatement, expressionStatement).WithTriviaFrom(ifStatement));
```

Comment:
>WithTriviaFrom [](http://example.com/codeflow?start=73&length=14)

was moving this to the wrapped block fixing a different bug?

Reply:
nope.  i just thought it read better.